Add `Beatmap tags` (#13171)
ppy/osu-wiki: wiki/Beatmap/Beatmap_tags/en.md

# Beatmap tags

*Not to be confused with [TAG difficulties](/wiki/Beatmap/TAG_beatmaps).*

> Keywords can be added to beatmaps to make searching for them more convenient. There are two types of tags, each with their own purpose.

Reviewer: ```suggestion
**Beatmap tags** are keywords that can be added to beatmaps to make searching for them more convenient. There are two types of tags, each with their own purpose.
```

## Mapper tags

*For rules regarding mapper tags, see: [Ranking criteria/Metadata § Tags](/wiki/Ranking_criteria/Metadata#tags)*

> **Mapper tags** can be used to include useful background information about a beatmap or song, e.g. the album name or the medium for which the song was produced for.

Reviewer: ```suggestion
**Mapper tags** are tags that describe useful background information about a beatmap or a song, e.g. the name of the album or the medium for which the song was produced for. As the name suggests, these tags are added by the beatmap host during the [ranking process](/wiki/Beatmap_ranking_procedure).
```

Author: Good idea mentioning the ranking process!

> When entering metadata in the [song setup window](/wiki/Client/Beatmap_editor/Song_setup), a beatmap host is able to choose custom keywords by separating them with spaces. Should a keyword itself contain spaces, they must be replaced with an underscore (`_`).

Reviewer: ```suggestion
When entering a beatmap's metadata in the [song setup window](/wiki/Client/Beatmap_editor/Song_setup), the beatmap host is able to add custom keywords to the `Tags` field by separating them with spaces. Keywords that consist of two or more words may be written with the spaces between them replaced with an underscore (`_`) to allow for a more efficient beatmap search.
```

Reviewer: (it's not a must)

> The keywords can be set as a filter in [song select](/wiki/Client/Interface#song-select) as well as the [beatmap listing](https://osu.ppy.sh/beatmapsets) via full-text search.

Reviewer: ```suggestion
These keywords can then be set as a filter in [song select](/wiki/Client/Interface#song-select) as well as the [beatmap listing](https://osu.ppy.sh/beatmapsets) via full-text search.
```

## User tags

> *Note: This feature is not available in osu!(stable).*

Reviewer: ```suggestion
*Note: The ability to add user tags in-game is only avalable in osu!(lazer).*
```

Author: I wanted to express here that **neither** searching nor **voting** for user tags is supported in osu!(stable).

This makes the sentence compared to my intention more ambigous in my opinion. This might also conflict with my concern [here](https://github.com/ppy/osu-wiki/pull/13171#discussion_r2051808924).

> **User tags** are keywords describing specific characteristics of a beatmap, such as patterns or required skill sets. After achieving at least a C [grade](/wiki/Gameplay/Grade) on a beatmap, users are able to select and vote for user tags in the result screen in osu!(lazer). Upon receiving at least 10 upvotes, the respective user tag will be displayed in the beatmap information page.

Reviewer: ```suggestion
**User tags** are tags that describe specific characteristics of a beatmap, such as its patterns or required skill sets. After achieving at least a C [grade](/wiki/Gameplay/Grade) on a beatmap, users are able to select and vote for user tags in the result screen in osu!(lazer). Upon receiving at least 10 upvotes, the respective user tag will be displayed in the beatmap information page.
```

> To filter beatmaps containing certain user tags, click them in the beatmap information page, or type in `tag="{name}"` where `{name}` is the name of any tag listed below.

Author: Not sure how I can phrase in a succinct way that searching for user tags can be done on osu-web (beatmap listing) and lazer (integrated beatmap listing).

> To filter beatmaps containing certain user tags, click them in the beatmap information page, or type in `tag="{name}"` where `{name}` is the name of any tag listed below.

Reviewer: ```suggestion
To filter for beatmaps with certain user tags, click on the corresponding tag in the beatmap information page or type in `tag="{name}"` where `{name}` is the name of any tag listed below.
```

Reviewer: i think this should be good enough

Author: ```suggestion
To filter for beatmaps with certain user tags, click on the corresponsing tag in the beatmap information page, or type in `tag="{name}"` in the beatmap listing where `{name}` is the name of any tag listed below.
```

> The tables below list all user tags grouped by category, including the game modes they are available in.

Reviewer: ```suggestion
The tables below list all user tags grouped by category as well as the game modes they are applicable in.
```

### Aim

| Tag name | Description | Game mode |
| :-- | :-- | :-- |

> | aim/aim control | Patterns with velocity or direction changes which strongly go against a player's natural movement pattern. | osu! |

Author: Should I remove the tag groups in the tables? Like so:

```suggestion
| aim control | Patterns with velocity or direction changes which strongly go against a player's natural movement pattern. | osu! |
```

- Mentioning them is not required for searching, and the search supports even full-text search (if the user tag contains no spaces, that is).
- Leaving them in adds a lot of visual clutter
- Would any tag names conflict with each other in this case? 🤔

Reviewer: i think this is fine as is

| aim/flow aim | Patterns with fully continuous cursor movement, usually due to a combination of wide angles and little time between objects. | osu! |
| aim/jump aim | Focuses heavily on jumps, i.e. circles spaced far apart that require the player to move towards, slow down to hit, then speed up to move towards the next object. | osu! |
| aim/linear aim | Patterns, usually jumps, that require the player to move continuously in a straight or nearly straight line. | osu! |
| aim/precision | Colloquial term for maps which require fine, precise movement to aim correctly. Typically refers to maps with circle sizes above and including 6. | osu! |

> | aim/precision | Colloquial term for maps which require fine, precise movement to catch fruits. Typically refers to maps with circle sizes above and including 6. | osu!catch |

Reviewer: `aim/precision` is written twice here

Author: Related to https://github.com/ppy/osu-wiki/pull/13171#issuecomment-2825326809

And to be clear, due to a tiny difference between the two, I have not combined them yet.

| tech/complex sv | Changes slider velocity by large amounts in order to test the player's reading and aim ability. | ![][osu!] |
| tech/complex sv | Changes slider velocity by large amounts in order to test the player's reading ability. | ![][osu!taiko] ![][osu!mania] |
| tech/flow | A map focused on natural and intuitive movement patterns. | ![][osu!catch] |
| tech/hyperwalks | A map which makes use of hyperdashes that require the player to walk, as otherwise, you will overshoot them. | ![][osu!catch] |
| tech/jump | A map that focuses on 1/2 constant dashes or hyperdashes. | ![][osu!catch] |
| tech/slider tech | Tests uncommon skills involving sliders, such as heavy use of kicksliders or aim on complex slidershapes. | ![][osu!] |
| tech/speed | A map that requires constant tapping at high BPMs. | ![][osu!] ![][osu!taiko] ![][osu!catch] ![][osu!mania] |
| tech/technical hybrid | Maps that feature the combination of both technical rice and Long Notes patterning. | ![][osu!mania] |
| tech/wiggles | A map with a focus on quick directional changes. | ![][osu!catch] |

> 

Author: ```suggestion

## Trivia

- The initial release of the user tags on 21 March 2025 was inspired from the [OMDB](https://omdb.nyahh.net/) project which had the similar goal of providing a user-curated tag system for beatmaps. Originally developed by ::{ flag=GB }:: [apollodw](https://osu.ppy.sh/users/9558549), the system was retired due to ...

```

On a quick search, I have not found a specific reason why the system was shutdown 🤔

Reviewer: the one running the website simply lost their interest apparently

http://x.com/bunnygirltwink/status/1892328084893815059

# Beatmap tags

*Not to be confused with [TAG difficulties](/wiki/Beatmap/TAG_beatmaps).*

> **Beatmap tags** are keywords that can be added to beatmaps to make searching for them more convenient. There are two types of tags, each with their own purpose.

Reviewer 2: ```suggestion
**Beatmap tags** are keywords that can be added to beatmaps to make [searching](/wiki/Beatmap_search) for them more convenient. There are two types of tags, each with their own purpose.
```

| aim/linear aim | Patterns, usually jumps, that require the player to move continuously in a straight or nearly straight line. | ![][osu!] |
| aim/precision | Colloquial term for maps which require fine, precise movement to aim correctly. Typically refers to maps with circle sizes above and including 6. | ![][osu!] |
| aim/precision | Colloquial term for maps which require fine, precise movement to catch fruits. Typically refers to maps with circle sizes above and including 6. | ![][osu!catch] |
| aim/sharp aim | Patterns with heavy use of sharp angle movement, either between circles or into sliders. | ![][osu!] |
| aim/spaced jumps | Focuses heavily on jumps with a lot of spacing. | ![][osu!] |
| aim/squares | Focuses heavily on jumps that follow square patterns. | ![][osu!] |
| aim/stamina | Tests a player's ability to aim spaced jumps over long periods of time. | ![][osu!] |
| aim/triangles | Focuses heavily on jumps that follow triangular patterns. | ![][osu!] |
| aim/wide aim | Patterns, usually jumps, with heavy use of wide angle movement. | ![][osu!] |

### Geometric

| Tag name | Description | Game mode |
| :-- | :-- | :-- |

> | geometric/grid snap | Objects are placed along a square grid, typically using osu!'s in-built grid snap feature. | ![][osu!] |

Reviewer 2: "built-in"?

Author: When writing the article, I asked this myself as well. I found [a thread](https://english.stackexchange.com/questions/129360/is-inbuilt-a-word-is-it-alright-to-use-it-or-should-i-use-built-in) stating that `in-built` is used in the UK.

 [Another thread](https://english.stackexchange.com/questions/1250/built-in-or-in-built) says that `built-in` is the much more common version of the two words.

Also related to https://github.com/ppy/osu-wiki/pull/13171#issuecomment-2825326809

Author: ```suggestion
| geometric/grid snap | Objects are placed along a square grid, typically using osu!'s built-in grid snap feature. | ![][osu!] |
```

| geometric/hexgrid | Objects are placed along a hexagonal grid. | ![][osu!] |

### Gimmick

| Tag name | Description | Game mode |
| :-- | :-- | :-- |
| gimmick/2B | Includes gameplay elements with two or more objects placed simultaneously. The term originates from a Chinese transliteration of 'idiot'. | ![][osu!] ![][osu!taiko] ![][osu!catch] ![][osu!mania] |
| gimmick/barlines | A map that makes use of barlines to enhance visuals or replace notes. | ![][osu!taiko] |
| gimmick/circle only | Restricts object choice to circles only. | ![][osu!] |
| gimmick/delay | Maps that feature high snap streams based on the song's delayed sound effect. | ![][osu!mania] |
| gimmick/dodge the beat | A map where the player needs to avoid every object. | ![][osu!catch] |

> | gimmick/LN inverse | Maps that feature long note holds that emphasize constant holds and releases in quick succession. Most distinct feature of Inverse is the "negative spaces" in the patterning. | ![][osu!mania] |

Reviewer 2: "emphasise"?

it seems these tag descriptions haven't quite been checked for british spelling. can handle that later

Author: Related to https://github.com/ppy/osu-wiki/pull/13171#issuecomment-2825326809

| tech/complex sv | Changes slider velocity by large amounts in order to test the player's reading and aim ability. | ![][osu!] |
| tech/complex sv | Changes slider velocity by large amounts in order to test the player's reading ability. | ![][osu!taiko] ![][osu!mania] |
| tech/flow | A map focused on natural and intuitive movement patterns. | ![][osu!catch] |
| tech/hyperwalks | A map which makes use of hyperdashes that require the player to walk, as otherwise, you will overshoot them. | ![][osu!catch] |
| tech/jump | A map that focuses on 1/2 constant dashes or hyperdashes. | ![][osu!catch] |
| tech/slider tech | Tests uncommon skills involving sliders, such as heavy use of kicksliders or aim on complex slidershapes. | ![][osu!] |
| tech/speed | A map that requires constant tapping at high BPMs. | ![][osu!] ![][osu!taiko] ![][osu!catch] ![][osu!mania] |
| tech/technical hybrid | Maps that feature the combination of both technical rice and Long Notes patterning. | ![][osu!mania] |
| tech/wiggles | A map with a focus on quick directional changes. | ![][osu!catch] |

## Trivia

> - The initial release of the user tags on 21 March 2025 was inspired from the [OMDB](https://omdb.nyahh.net/) project which had the similar goal of providing a user-curated tag system for beatmaps. Originally developed by ::{ flag=GB }:: [apollodw](https://osu.ppy.sh/users/9558549), the project was retired on 1 March 2025 due to personal reasons.

Author: ```suggestion
- The initial release of the user tags on 21 March 2025 was inspired from the [OMDB](https://omdb.nyahh.net/) project which had the similar goal of providing a user-curated tag system for beatmaps. Originally developed by ::{ flag=GB }:: [apollodw](https://osu.ppy.sh/users/9558549), the project was retired on 1 March 2025 [due to personal reasons](https://x.com/bunnygirltwink/status/1892328084893815059).
```
